src/kfs_frontend/kfs_grpc_inference_service.cpp

```diff
@@ -257,7 +281,7 @@ Status KFSInferenceServiceImpl::ModelInferImpl(::grpc::ServerContext* context, c
     if (pipelinePtr) {
         reporterOut = &pipelinePtr->getMetricReporter();
         status = pipelinePtr->execute(executionContext);
-    } else {
+    } else if (modelInstance) {
         reporterOut = &modelInstance->getMetricReporter();
         status = modelInstance->infer(request, response, modelInstanceUnloadGuard);
     }
```

Comment:
Here we should check if reporterOut is defined, it will not be defined if pipelinePtr and modelInstance are both null. Add:
```
} else {
  throw ?
}
```